app/src/main/java/io/snabble/pay/app/feature/detailsaccount/ui/AccountDetails.kt

```diff
@@ -46,77 +51,113 @@ fun AccountDetails(
     onLabelChange: (label: String, colors: List<String>) -> Unit,
     onDeleteAccount: () -> Unit,
 ) {
+    val showChangeTitleDialog = rememberSaveable {
+        mutableStateOf(false)
+    }
+
     val cardName = rememberSaveable(inputs = arrayOf(accountCard.name)) {
         mutableStateOf(accountCard.name)
     }
-
     val showDeleteDialog = remember { mutableStateOf(false) }
-
-    AppBarLayout(
-        title = "",
-        icon = Icons.Filled.Clear,
-        onBackClick = {
-            navigator?.navigate(HomeScreenDestination)
-        }
-    ) {
-        if (showDeleteDialog.value) {
-            SnabblePayDialog(
-                dialogTitle = stringResource(id = R.string.delete_card_title),
-                dialogText = stringResource(id = R.string.delete_card_message),
-                confirmButtonLabel = stringResource(id = R.string.delete_card_confirm),
-                onConfirm = {
-                    onDeleteAccount()
-                    showDeleteDialog.value = false
-                },
-                cancelButtonLabel = stringResource(id = R.string.delete_card_cancel)
-            ) { showDeleteDialog.value = false }
-        }
-        Column(modifier = Modifier.fillMaxWidth()) {
-            Spacer(modifier = Modifier.height(16.dp))
-            EditTextFieldCentered(
-                modifier = Modifier
-                    .fillMaxWidth()
-                    .padding(horizontal = 16.dp),
-                value = cardName.value,
-                onValueChange = { cardName.value = it },
-                onAction = { onLabelChange(cardName.value, accountCard.cardBackgroundColor) }
-            )
-            Spacer(modifier = Modifier.height(16.dp))
-            Box(modifier = Modifier.fillMaxWidth()) {
-                DetailsBackground(modifier = Modifier.fillMaxWidth())
-                Column(
-                    modifier = Modifier
-                        .fillMaxSize()
-                        .padding(start = 16.dp, end = 16.dp, bottom = 16.dp)
+    SnabblePayTheme {
+        AppBarLayout(
+            title = cardName.value,
+            onBackClick = {
+                navigator?.navigate(HomeScreenDestination)
+            }
+        ) {
+            if (showChangeTitleDialog.value) {
+                SnabblePayDialog(
+                    dialogTitle = stringResource(id = R.string.account_details_change_title),
+                    secondaryButtonLabel = stringResource(id = R.string.change_title_dialog_button),
+                    onDismiss = {
+                        onLabelChange(cardName.value, accountCard.cardBackgroundColor)
+                        showChangeTitleDialog.value = false
+                    }
```

Comment:
The change dialog now has the problem, that the if the secondary button is used for the primary action, a dismiss of the dialog is also like confirming. That is uncommon and should not be the case. I guess the intention was to use the style from the secondary button as primary action, like in the design, but I'd recommend sorting this out with UI/UX that the primary action button should appear in the primary action style.

Reply:
changed

Reply:
There is no change yet!

Reply:
Oh i forgot to push...